libs/sdk-backend-spi/src/workspace/attributes/index.ts

```diff
@@ -1,12 +1,28 @@
-// (C) 2019-2022 GoodData Corporation
+// (C) 2019-2025 GoodData Corporation
 import {
     ObjRef,
     IAttributeDisplayFormMetadataObject,
     IAttributeMetadataObject,
     IMetadataObject,
+    IDataSetMetadataObject,
 } from "@gooddata/sdk-model";
 import { IElementsQueryFactory } from "./elements/index.js";
 
+/**
+ * @beta
+ */
+export interface IAttributeWithReferences {
+    /**
+     * Attribute metadata object
+     */
+    attribute: IAttributeMetadataObject;
+
+    /**
+     * Attribute related data set metadata object.
+     */
+    dataSet?: IDataSetMetadataObject;
```

Comment:
I'm just curious why the dataSet info was not put under IAttributeMetadataObject, similarly as other include type which is displayForms. Is it because having it inside would complicate to distinguish where the attribute come from if it would be just optional there, like it's optional in IAttributeWithReferences?

Reply:
We consider this later, postponing it now.